src/Helpers/AuditResolver.php

```diff
@@ -9,13 +9,19 @@
 
 class AuditResolver
 {
-    public static function findOwner(?Trackable $model): ?Model
+    public static function findOwner(?Trackable $trackable): ?Model
     {
-        if (empty($model)) {
+        if (empty($trackable)) {
             return null;
         }
 
-        $owner = $model->trackableOwnerResolver();
+        $ownerRelationshipName = $trackable->getOwnerRelationshipName();
+
+        if ($ownerRelationshipName && $trackable instanceof Model) {
+            $trackable->loadMissing($ownerRelationshipName);
+        }
```

Comment:
This is still technically running a query for each. I would look into a way to figure out if we can eager load via a `with` or a load on a collection instead.

Reply:
Discussed a solution to this, using Laravel's `morphWith` on the `MorphTo` class [[docs](https://laravel.com/docs/11.x/eloquent-relationships#nested-eager-loading-morphto-relationships)] to load relationships using the `getOwnerRelationshipName` function that must be defined on tracked models in the logger config.